src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -38,19 +38,21 @@ public static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Pref
     }
 
     /**
+     * Parses a String input {@code entity} into a normalized String entity.
      * Throws ParseException if {@code entity} provided is not a valid entity in the address book.
      * Leading and trailing whitespaces will be trimmed.
      */
-    public static void requireValidEntity(String entity) throws ParseException {
+    public static String parseEntity(String entity) throws ParseException {
         requireNonNull(entity);
         String trimmedEntity = entity.trim();
+        String normalizedEntity = trimmedEntity.toLowerCase();
 
         // TODO: Not sure if this is magic string?
-        switch (trimmedEntity) {
+        switch (normalizedEntity) {
         case "contact", "company", "job", "all":
```

Comment:
Will this cause an unexpected error message if say I put add all?

Reply:
This wont, because as far this this function is concern, that is a valid entity. 

In my current implementation, `add all` will throw operation not allowed error in the Addparser switch case.

![image](https://github.com/user-attachments/assets/346bc7de-e822-4b15-9335-e6acbf90a49c)

Reply:
Alright then that's good.